### PR TITLE
Fixes issue #880: Support carrying the precomputed java inspections in the DataShape specification field.

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

File: app/ui/src/app/integrations/edit-page/step-configure/data-mapper/data-mapper-host.component.ts [hunk 1]
```diff
@@ -108,31 +108,37 @@ export class DataMapperHostComponent extends FlowPage implements OnInit {
...
-            log.warnc(
-              () =>
-                'No precomputed java document found for ' + type + ': ' + err,
-              category
-            );
-            this.removeInitializationTask();
-          }
-        );
+
+        if ( specification != '' ) {
```
Comment (deeleman) [inline app/ui/src/app/integrations/edit-page/step-configure/data-mapper/data-mapper-host.component.ts]: I checked this yesterday and apparently out linter is skipping these constructs, although it's configured to block them. Basically  we do not allow extra spaces inside parens anymore in function or conditional expressions anymore.

File: app/ui/src/app/integrations/edit-page/step-configure/data-mapper/data-mapper-host.component.ts [hunk 2]
```diff
@@ -108,31 +108,37 @@ export class DataMapperHostComponent extends FlowPage implements OnInit {
...
+              );
+              log.debugc(() => inspection, category);
+              docDef.initCfg.inspectionResultContents = inspection;
+              this.removeInitializationTask();
+            },
+            err => {
+              log.warnc(
+                () =>
+                  'No precomputed java document found for ' + type + ': ' + err,
+                category
```
Comment (deeleman) [inline app/ui/src/app/integrations/edit-page/step-configure/data-mapper/data-mapper-host.component.ts]: The linter is also configured to complain when indenting mismatches are detected in the same code block, but dunno if this one would get detected. Just for the sake of readability, I'd suggest to align both the string with the `category` variable in the lambda payload. BTW, when chaining two or more strings and variables, ES6 templated strings are the way to go:

```javascript
log.warnc(
  () =>
    `No precomputed java document found for ${type}: ${err}`,
    category
);
```

File: app/ui/src/app/integrations/edit-page/step-configure/data-mapper/data-mapper-host.component.ts (hunk 1 above)
Comment (chirino) [reply app/ui/src/app/integrations/edit-page/step-configure/data-mapper/data-mapper-host.component.ts]: ms code marks it red without the spaces.

File: app/ui/src/app/integrations/edit-page/step-configure/data-mapper/data-mapper-host.component.ts (hunk 2 above)
Comment (chirino) [reply app/ui/src/app/integrations/edit-page/step-configure/data-mapper/data-mapper-host.component.ts]: will do next time.  Good tip.

File: app/ui/src/app/integrations/edit-page/step-configure/data-mapper/data-mapper-host.component.ts (hunk 1 above)
Comment (deeleman) [reply app/ui/src/app/integrations/edit-page/step-configure/data-mapper/data-mapper-host.component.ts]: It shouldn't. That seems to be caused by your visual style/linting config or perhaps you have the Angular Language Services plugin installed, which sometimes feature an erratic behavior when validating code.

**Comment by gashcrumb** [comment]
Just needs a rebase now...

**Pull request merged**